swarm/network/kademlia.go

```diff
@@ -442,6 +452,49 @@ func (k *Kademlia) ListKnown() []*BzzAddr {
 	return res
 }
 
+func (k *Kademlia) Nearest(chunkRef string) (string, error) {
+	b, err := hex.DecodeString(chunkRef)
+	if err != nil {
+		return "", err
+	}
+
+	var sp *Peer
+
+	k.EachConn(b, 255, func(p *Peer, po int) bool {
+		// skip light nodes
+		if p.LightNode {
+			return true
```

Comment:
i dont understand this... the cryptic eachConn code might suggest that returning `true` keeps the iterator going, in which case this statement would be correct, but the one at L471 would stop the iterator after 1 iteration (given that the first result is not a light node)

Reply:
All our Kademlia code is cryptic. I will remove all these API calls, as I don't want us to discuss this now.